server/gameObjects/Player.mjs

```diff
@@ -402,8 +402,13 @@ export default class Player {
   }
 
   canGetResource(energyCost) {
-    // @TODO: send message that action can't be performed #164
-    return this.isInRange(1) && this.energy >= energyCost;
+    if (!this.isInRange(1)) {
+      return getMsgKey(MESSAGES.not_in_range);
```

Comment:
still, I would prefer to have something like
`MESSAGE_TYPES.NOT_IN_RANGE`
here.
1. you won't need to import MESSAGES and function here, only MESSAGES_TYPES
2. that function in my opinion is not needed complication. the only thing you need to do is find the property in the object. And we can achieve that by reading the property from the object which is O(1) complexity and Object.keys().find() is just not necessary